### ISSUE TITLE
Deploying js-ipfs to the Galaxy

### ISSUE THREAD

**Issue by diasdavid** [ipfs/js-ipfs#804]
Similar to go-ipfs, we want to run some js-ipfs nodes and monitor their behaviour. From a quick chat with @lgierth on IRC, here are some notes:

```
14:48 <@lgierth> daviddias: about running js-ipfs, would be best if we can have a dockerfile, and metrics exposed at :5001/debug/metrics/prometheus
14:49 <@lgierth> like, number of peers and such stuff
14:49 <@lgierth> datastore stats
14:50 <@daviddias> lgierth: thanks, that is what I needed to know :) 
14:50 <@daviddias> will let you know once we have metrics exposed, right now we have none
14:52 <@lgierth> memory/cpu usage and gc timings are welcome too but can at least mem/cpu can be pulled out of docker too
14:53 <@lgierth> relays?
15:00 <@daviddias> lgierth: is there a format that you expect coming from it?
15:06 <@lgierth> yeah prometheus text format
15:06 <@lgierth> let me see
15:07 <@lgierth> https://prometheus.io/docs/instrumenting/writing_exporters/
15:07 <@lgierth> or https://github.com/siimon/prom-client if it lgty
```

**Comment by dignifiedquire** [comment]
![nasa-89125](https://cloud.githubusercontent.com/assets/790842/24155338/dbb1ad8e-0e53-11e7-9f2f-c59eae184812.jpg)

**Comment by dryajov** [comment]
@lgierth @diasdavid 

I have a not there regarding passive and active relays:
> All peers are relays, but they can function in two modes:
> * active/proactive relays that dial into other nodes
>     * this are explicitly enabled by config options
> * passive relays, which only relay over known connections, no explicit dialing. If no connection exists or the requested peer, the relay is refused.

How are active relays gonna be handled, does it make sense to have them as dedicated nodes, maybe part of the bootstrap list?

https://github.com/libp2p/js-libp2p-circuit/issues/4

**Comment by VictorBjelkholm** [comment]
# Metrics to expose

- Connected peers
- Memory usage
- CPU usage
- GC stats
- Datastore stats ( @diasdavid can you specify which? )

**Comment by diasdavid** [comment]
@VictorBjelkholm is Datastore or Repo? Otherwise looks good :)

**Comment by VictorBjelkholm** [comment]
@diasdavid 
> is Datastore or Repo? Otherwise looks good :)

Probably repository on second thought, went by the IRC log posted in the beginning.

> 14:49 <@lgierth> datastore stats

For reference, this is a example call to `/debug/metrics/prometheus` on go-ipfs:

<details>

```

# HELP go_gc_duration_seconds A summary of the GC invocation durations.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 1.9872e-05
go_gc_duration_seconds{quantile="0.25"} 4.9813e-05
go_gc_duration_seconds{quantile="0.5"} 7.7594e-05
go_gc_duration_seconds{quantile="0.75"} 0.000117754
go_gc_duration_seconds{quantile="1"} 0.000215895
go_gc_duration_seconds_sum 0.000537014
go_gc_duration_seconds_count 6
# HELP go_goroutines Number of goroutines that currently exist.
# TYPE go_goroutines gauge
go_goroutines 103
# HELP go_memstats_alloc_bytes Number of bytes allocated and still in use.
# TYPE go_memstats_alloc_bytes gauge
go_memstats_alloc_bytes 1.5215592e+07
# HELP go_memstats_alloc_bytes_total Total number of bytes allocated, even if freed.
# TYPE go_memstats_alloc_bytes_total counter
go_memstats_alloc_bytes_total 2.9086096e+07
# HELP go_memstats_buck_hash_sys_bytes Number of bytes used by the profiling bucket hash table.
# TYPE go_memstats_buck_hash_sys_bytes gauge
go_memstats_buck_hash_sys_bytes 1.456261e+06
# HELP go_memstats_frees_total Total number of frees.
# TYPE go_memstats_frees_total counter
go_memstats_frees_total 66795
# HELP go_memstats_gc_sys_bytes Number of bytes used for garbage collection system metadata.
# TYPE go_memstats_gc_sys_bytes gauge
go_memstats_gc_sys_bytes 827392
# HELP go_memstats_heap_alloc_bytes Number of heap bytes allocated and still in use.
# TYPE go_memstats_heap_alloc_bytes gauge
go_memstats_heap_alloc_bytes 1.5215592e+07
# HELP go_memstats_heap_idle_bytes Number of heap bytes waiting to be used.
# TYPE go_memstats_heap_idle_bytes gauge
go_memstats_heap_idle_bytes 229376
# HELP go_memstats_heap_inuse_bytes Number of heap bytes that are in use.
# TYPE go_memstats_heap_inuse_bytes gauge
go_memstats_heap_inuse_bytes 1.6351232e+07
# HELP go_memstats_heap_objects Number of allocated objects.
# TYPE go_memstats_heap_objects gauge
go_memstats_heap_objects 99595
# HELP go_memstats_heap_released_bytes_total Total number of heap bytes released to OS.
# TYPE go_memstats_heap_released_bytes_total counter
go_memstats_heap_released_bytes_total 0
# HELP go_memstats_heap_sys_bytes Number of heap bytes obtained from system.
# TYPE go_memstats_heap_sys_bytes gauge
go_memstats_heap_sys_bytes 1.6580608e+07
# HELP go_memstats_last_gc_time_seconds Number of seconds since 1970 of last garbage collection.
# TYPE go_memstats_last_gc_time_seconds gauge
go_memstats_last_gc_time_seconds 1.4993547201908672e+09
# HELP go_memstats_lookups_total Total number of pointer lookups.
# TYPE go_memstats_lookups_total counter
go_memstats_lookups_total 114
# HELP go_memstats_mallocs_total Total number of mallocs.
# TYPE go_memstats_mallocs_total counter
go_memstats_mallocs_total 166390
# HELP go_memstats_mcache_inuse_bytes Number of bytes in use by mcache structures.
# TYPE go_memstats_mcache_inuse_bytes gauge
go_memstats_mcache_inuse_bytes 4800
# HELP go_memstats_mcache_sys_bytes Number of bytes used for mcache structures obtained from system.
# TYPE go_memstats_mcache_sys_bytes gauge
go_memstats_mcache_sys_bytes 16384
# HELP go_memstats_mspan_inuse_bytes Number of bytes in use by mspan structures.
# TYPE go_memstats_mspan_inuse_bytes gauge
go_memstats_mspan_inuse_bytes 145160
# HELP go_memstats_mspan_sys_bytes Number of bytes used for mspan structures obtained from system.
# TYPE go_memstats_mspan_sys_bytes gauge
go_memstats_mspan_sys_bytes 147456
# HELP go_memstats_next_gc_bytes Number of heap bytes when next garbage collection will take place.
# TYPE go_memstats_next_gc_bytes gauge
go_memstats_next_gc_bytes 1.8126784e+07
# HELP go_memstats_other_sys_bytes Number of bytes used for other system allocations.
# TYPE go_memstats_other_sys_bytes gauge
go_memstats_other_sys_bytes 1.142899e+06
# HELP go_memstats_stack_inuse_bytes Number of bytes in use by the stack allocator.
# TYPE go_memstats_stack_inuse_bytes gauge
go_memstats_stack_inuse_bytes 1.245184e+06
# HELP go_memstats_stack_sys_bytes Number of bytes obtained from system for stack allocator.
# TYPE go_memstats_stack_sys_bytes gauge
go_memstats_stack_sys_bytes 1.245184e+06
# HELP go_memstats_sys_bytes Number of bytes obtained by system. Sum of all system allocations.
# TYPE go_memstats_sys_bytes gauge
go_memstats_sys_bytes 2.1416184e+07
# HELP http_request_duration_microseconds The HTTP request latencies in microseconds.
# TYPE http_request_duration_microseconds summary
http_request_duration_microseconds{handler="api",quantile="0.5"} 9.123
http_request_duration_microseconds{handler="api",quantile="0.9"} 11.684
http_request_duration_microseconds{handler="api",quantile="0.99"} 11.684
http_request_duration_microseconds_sum{handler="api"} 69.877
http_request_duration_microseconds_count{handler="api"} 6
http_request_duration_microseconds{handler="gateway",quantile="0.5"} NaN
http_request_duration_microseconds{handler="gateway",quantile="0.9"} NaN
http_request_duration_microseconds{handler="gateway",quantile="0.99"} NaN
http_request_duration_microseconds_sum{handler="gateway"} 0
http_request_duration_microseconds_count{handler="gateway"} 0
# HELP http_request_size_bytes The HTTP request sizes in bytes.
# TYPE http_request_size_bytes summary
http_request_size_bytes{handler="api",quantile="0.5"} 67
http_request_size_bytes{handler="api",quantile="0.9"} 72
http_request_size_bytes{handler="api",quantile="0.99"} 72
http_request_size_bytes_sum{handler="api"} 406
http_request_size_bytes_count{handler="api"} 6
http_request_size_bytes{handler="gateway",quantile="0.5"} NaN
http_request_size_bytes{handler="gateway",quantile="0.9"} NaN
http_request_size_bytes{handler="gateway",quantile="0.99"} NaN
http_request_size_bytes_sum{handler="gateway"} 0
http_request_size_bytes_count{handler="gateway"} 0
# HELP http_requests_total Total number of HTTP requests made.
# TYPE http_requests_total counter
http_requests_total{code="404",handler="api",method="get"} 6
# HELP http_response_size_bytes The HTTP response sizes in bytes.
# TYPE http_response_size_bytes summary
http_response_size_bytes{handler="api",quantile="0.5"} 19
http_response_size_bytes{handler="api",quantile="0.9"} 19
http_response_size_bytes{handler="api",quantile="0.99"} 19
http_response_size_bytes_sum{handler="api"} 114
http_response_size_bytes_count{handler="api"} 6
http_response_size_bytes{handler="gateway",quantile="0.5"} NaN
http_response_size_bytes{handler="gateway",quantile="0.9"} NaN
http_response_size_bytes{handler="gateway",quantile="0.99"} NaN
http_response_size_bytes_sum{handler="gateway"} 0
http_response_size_bytes_count{handler="gateway"} 0
# HELP ipfs_bitswap_recv_all_blocks_bytes Summary of all data blocks recived
# TYPE ipfs_bitswap_recv_all_blocks_bytes histogram
ipfs_bitswap_recv_all_blocks_bytes_bucket{le="64"} 0
ipfs_bitswap_recv_all_blocks_bytes_bucket{le="1024"} 0
ipfs_bitswap_recv_all_blocks_bytes_bucket{le="16384"} 0
ipfs_bitswap_recv_all_blocks_bytes_bucket{le="262144"} 0
ipfs_bitswap_recv_all_blocks_bytes_bucket{le="262159"} 0
ipfs_bitswap_recv_all_blocks_bytes_bucket{le="4.194304e+06"} 0
ipfs_bitswap_recv_all_blocks_bytes_bucket{le="+Inf"} 0
ipfs_bitswap_recv_all_blocks_bytes_sum 0
ipfs_bitswap_recv_all_blocks_bytes_count 0
# HELP ipfs_bitswap_recv_dup_blocks_bytes Summary of duplicate data blocks recived
# TYPE ipfs_bitswap_recv_dup_blocks_bytes histogram
ipfs_bitswap_recv_dup_blocks_bytes_bucket{le="64"} 0
ipfs_bitswap_recv_dup_blocks_bytes_bucket{le="1024"} 0
ipfs_bitswap_recv_dup_blocks_bytes_bucket{le="16384"} 0
ipfs_bitswap_recv_dup_blocks_bytes_bucket{le="262144"} 0
ipfs_bitswap_recv_dup_blocks_bytes_bucket{le="262159"} 0
ipfs_bitswap_recv_dup_blocks_bytes_bucket{le="4.194304e+06"} 0
ipfs_bitswap_recv_dup_blocks_bytes_bucket{le="+Inf"} 0
ipfs_bitswap_recv_dup_blocks_bytes_sum 0
ipfs_bitswap_recv_dup_blocks_bytes_count 0
# HELP ipfs_bitswap_sent_all_blocks_bytes Histogram of blocks sent by this bitswap
# TYPE ipfs_bitswap_sent_all_blocks_bytes histogram
ipfs_bitswap_sent_all_blocks_bytes_bucket{le="64"} 0
ipfs_bitswap_sent_all_blocks_bytes_bucket{le="1024"} 0
ipfs_bitswap_sent_all_blocks_bytes_bucket{le="16384"} 0
ipfs_bitswap_sent_all_blocks_bytes_bucket{le="262144"} 0
ipfs_bitswap_sent_all_blocks_bytes_bucket{le="262159"} 0
ipfs_bitswap_sent_all_blocks_bytes_bucket{le="4.194304e+06"} 0
ipfs_bitswap_sent_all_blocks_bytes_bucket{le="+Inf"} 0
ipfs_bitswap_sent_all_blocks_bytes_sum 0
ipfs_bitswap_sent_all_blocks_bytes_count 0
# HELP ipfs_bitswap_wantlist_total Number of items in wantlist.
# TYPE ipfs_bitswap_wantlist_total gauge
ipfs_bitswap_wantlist_total 0
# HELP ipfs_bs_cache_arc_hits_total Number of ARC cache hits
# TYPE ipfs_bs_cache_arc_hits_total counter
ipfs_bs_cache_arc_hits_total 9
# HELP ipfs_bs_cache_arc_total Total number of ARC cache requests
# TYPE ipfs_bs_cache_arc_total counter
ipfs_bs_cache_arc_total 27
# HELP ipfs_fsrepo_datastore_blocks_delete_errors_total Number of errored Datastore.Delete calls
# TYPE ipfs_fsrepo_datastore_blocks_delete_errors_total counter
ipfs_fsrepo_datastore_blocks_delete_errors_total 0
# HELP ipfs_fsrepo_datastore_blocks_delete_latency_seconds Latency distribution of Datastore.Delete calls
# TYPE ipfs_fsrepo_datastore_blocks_delete_latency_seconds histogram
ipfs_fsrepo_datastore_blocks_delete_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_blocks_delete_latency_seconds_bucket{le="0.001"} 0
ipfs_fsrepo_datastore_blocks_delete_latency_seconds_bucket{le="0.01"} 0
ipfs_fsrepo_datastore_blocks_delete_latency_seconds_bucket{le="0.1"} 0
ipfs_fsrepo_datastore_blocks_delete_latency_seconds_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_blocks_delete_latency_seconds_sum 0
ipfs_fsrepo_datastore_blocks_delete_latency_seconds_count 0
# HELP ipfs_fsrepo_datastore_blocks_delete_total Total number of Datastore.Delete calls
# TYPE ipfs_fsrepo_datastore_blocks_delete_total counter
ipfs_fsrepo_datastore_blocks_delete_total 0
# HELP ipfs_fsrepo_datastore_blocks_get_errors_total Number of errored Datastore.Get calls
# TYPE ipfs_fsrepo_datastore_blocks_get_errors_total counter
ipfs_fsrepo_datastore_blocks_get_errors_total 0
# HELP ipfs_fsrepo_datastore_blocks_get_latency_seconds Latency distribution of Datastore.Get calls
# TYPE ipfs_fsrepo_datastore_blocks_get_latency_seconds histogram
ipfs_fsrepo_datastore_blocks_get_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_blocks_get_latency_seconds_bucket{le="0.001"} 3
ipfs_fsrepo_datastore_blocks_get_latency_seconds_bucket{le="0.01"} 4
ipfs_fsrepo_datastore_blocks_get_latency_seconds_bucket{le="0.1"} 4
ipfs_fsrepo_datastore_blocks_get_latency_seconds_bucket{le="+Inf"} 4
ipfs_fsrepo_datastore_blocks_get_latency_seconds_sum 0.0020888639999999997
ipfs_fsrepo_datastore_blocks_get_latency_seconds_count 4
# HELP ipfs_fsrepo_datastore_blocks_get_size_bytes Size distribution of retrieved byte slices
# TYPE ipfs_fsrepo_datastore_blocks_get_size_bytes histogram
ipfs_fsrepo_datastore_blocks_get_size_bytes_bucket{le="64"} 1
ipfs_fsrepo_datastore_blocks_get_size_bytes_bucket{le="4096"} 2
ipfs_fsrepo_datastore_blocks_get_size_bytes_bucket{le="262144"} 4
ipfs_fsrepo_datastore_blocks_get_size_bytes_bucket{le="1.6777216e+07"} 4
ipfs_fsrepo_datastore_blocks_get_size_bytes_bucket{le="+Inf"} 4
ipfs_fsrepo_datastore_blocks_get_size_bytes_sum 24485
ipfs_fsrepo_datastore_blocks_get_size_bytes_count 4
# HELP ipfs_fsrepo_datastore_blocks_get_total Total number of Datastore.Get calls
# TYPE ipfs_fsrepo_datastore_blocks_get_total counter
ipfs_fsrepo_datastore_blocks_get_total 4
# HELP ipfs_fsrepo_datastore_blocks_has_errors_total Number of errored Datastore.Has calls
# TYPE ipfs_fsrepo_datastore_blocks_has_errors_total counter
ipfs_fsrepo_datastore_blocks_has_errors_total 0
# HELP ipfs_fsrepo_datastore_blocks_has_latency_seconds Latency distribution of Datastore.Has calls
# TYPE ipfs_fsrepo_datastore_blocks_has_latency_seconds histogram
ipfs_fsrepo_datastore_blocks_has_latency_seconds_bucket{le="0.0001"} 9
ipfs_fsrepo_datastore_blocks_has_latency_seconds_bucket{le="0.001"} 10
ipfs_fsrepo_datastore_blocks_has_latency_seconds_bucket{le="0.01"} 13
ipfs_fsrepo_datastore_blocks_has_latency_seconds_bucket{le="0.1"} 14
ipfs_fsrepo_datastore_blocks_has_latency_seconds_bucket{le="+Inf"} 14
ipfs_fsrepo_datastore_blocks_has_latency_seconds_sum 0.047042794
ipfs_fsrepo_datastore_blocks_has_latency_seconds_count 14
# HELP ipfs_fsrepo_datastore_blocks_has_total Total number of Datastore.Has calls
# TYPE ipfs_fsrepo_datastore_blocks_has_total counter
ipfs_fsrepo_datastore_blocks_has_total 14
# HELP ipfs_fsrepo_datastore_blocks_put_errors_total Number of errored Datastore.Put calls
# TYPE ipfs_fsrepo_datastore_blocks_put_errors_total counter
ipfs_fsrepo_datastore_blocks_put_errors_total 0
# HELP ipfs_fsrepo_datastore_blocks_put_latency_seconds Latency distribution of Datastore.Put calls
# TYPE ipfs_fsrepo_datastore_blocks_put_latency_seconds histogram
ipfs_fsrepo_datastore_blocks_put_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_blocks_put_latency_seconds_bucket{le="0.001"} 0
ipfs_fsrepo_datastore_blocks_put_latency_seconds_bucket{le="0.01"} 0
ipfs_fsrepo_datastore_blocks_put_latency_seconds_bucket{le="0.1"} 0
ipfs_fsrepo_datastore_blocks_put_latency_seconds_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_blocks_put_latency_seconds_sum 0
ipfs_fsrepo_datastore_blocks_put_latency_seconds_count 0
# HELP ipfs_fsrepo_datastore_blocks_put_size_bytes Size distribution of stored byte slices
# TYPE ipfs_fsrepo_datastore_blocks_put_size_bytes histogram
ipfs_fsrepo_datastore_blocks_put_size_bytes_bucket{le="64"} 0
ipfs_fsrepo_datastore_blocks_put_size_bytes_bucket{le="4096"} 0
ipfs_fsrepo_datastore_blocks_put_size_bytes_bucket{le="262144"} 0
ipfs_fsrepo_datastore_blocks_put_size_bytes_bucket{le="1.6777216e+07"} 0
ipfs_fsrepo_datastore_blocks_put_size_bytes_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_blocks_put_size_bytes_sum 0
ipfs_fsrepo_datastore_blocks_put_size_bytes_count 0
# HELP ipfs_fsrepo_datastore_blocks_put_total Total number of Datastore.Put calls
# TYPE ipfs_fsrepo_datastore_blocks_put_total counter
ipfs_fsrepo_datastore_blocks_put_total 0
# HELP ipfs_fsrepo_datastore_blocks_query_errors_total Number of errored Datastore.Query calls
# TYPE ipfs_fsrepo_datastore_blocks_query_errors_total counter
ipfs_fsrepo_datastore_blocks_query_errors_total 0
# HELP ipfs_fsrepo_datastore_blocks_query_latency_seconds Latency distribution of Datastore.Query calls
# TYPE ipfs_fsrepo_datastore_blocks_query_latency_seconds histogram
ipfs_fsrepo_datastore_blocks_query_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_blocks_query_latency_seconds_bucket{le="0.001"} 0
ipfs_fsrepo_datastore_blocks_query_latency_seconds_bucket{le="0.01"} 0
ipfs_fsrepo_datastore_blocks_query_latency_seconds_bucket{le="0.1"} 0
ipfs_fsrepo_datastore_blocks_query_latency_seconds_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_blocks_query_latency_seconds_sum 0
ipfs_fsrepo_datastore_blocks_query_latency_seconds_count 0
# HELP ipfs_fsrepo_datastore_blocks_query_total Total number of Datastore.Query calls
# TYPE ipfs_fsrepo_datastore_blocks_query_total counter
ipfs_fsrepo_datastore_blocks_query_total 0
# HELP ipfs_fsrepo_datastore_delete_errors_total Number of errored Datastore.Delete calls
# TYPE ipfs_fsrepo_datastore_delete_errors_total counter
ipfs_fsrepo_datastore_delete_errors_total 0
# HELP ipfs_fsrepo_datastore_delete_latency_seconds Latency distribution of Datastore.Delete calls
# TYPE ipfs_fsrepo_datastore_delete_latency_seconds histogram
ipfs_fsrepo_datastore_delete_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_delete_latency_seconds_bucket{le="0.001"} 0
ipfs_fsrepo_datastore_delete_latency_seconds_bucket{le="0.01"} 0
ipfs_fsrepo_datastore_delete_latency_seconds_bucket{le="0.1"} 0
ipfs_fsrepo_datastore_delete_latency_seconds_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_delete_latency_seconds_sum 0
ipfs_fsrepo_datastore_delete_latency_seconds_count 0
# HELP ipfs_fsrepo_datastore_delete_total Total number of Datastore.Delete calls
# TYPE ipfs_fsrepo_datastore_delete_total counter
ipfs_fsrepo_datastore_delete_total 0
# HELP ipfs_fsrepo_datastore_get_errors_total Number of errored Datastore.Get calls
# TYPE ipfs_fsrepo_datastore_get_errors_total counter
ipfs_fsrepo_datastore_get_errors_total 0
# HELP ipfs_fsrepo_datastore_get_latency_seconds Latency distribution of Datastore.Get calls
# TYPE ipfs_fsrepo_datastore_get_latency_seconds histogram
ipfs_fsrepo_datastore_get_latency_seconds_bucket{le="0.0001"} 2
ipfs_fsrepo_datastore_get_latency_seconds_bucket{le="0.001"} 5
ipfs_fsrepo_datastore_get_latency_seconds_bucket{le="0.01"} 6
ipfs_fsrepo_datastore_get_latency_seconds_bucket{le="0.1"} 6
ipfs_fsrepo_datastore_get_latency_seconds_bucket{le="+Inf"} 6
ipfs_fsrepo_datastore_get_latency_seconds_sum 0.002168916
ipfs_fsrepo_datastore_get_latency_seconds_count 6
# HELP ipfs_fsrepo_datastore_get_size_bytes Size distribution of retrieved byte slices
# TYPE ipfs_fsrepo_datastore_get_size_bytes histogram
ipfs_fsrepo_datastore_get_size_bytes_bucket{le="64"} 3
ipfs_fsrepo_datastore_get_size_bytes_bucket{le="4096"} 4
ipfs_fsrepo_datastore_get_size_bytes_bucket{le="262144"} 6
ipfs_fsrepo_datastore_get_size_bytes_bucket{le="1.6777216e+07"} 6
ipfs_fsrepo_datastore_get_size_bytes_bucket{le="+Inf"} 6
ipfs_fsrepo_datastore_get_size_bytes_sum 24553
ipfs_fsrepo_datastore_get_size_bytes_count 6
# HELP ipfs_fsrepo_datastore_get_total Total number of Datastore.Get calls
# TYPE ipfs_fsrepo_datastore_get_total counter
ipfs_fsrepo_datastore_get_total 6
# HELP ipfs_fsrepo_datastore_has_errors_total Number of errored Datastore.Has calls
# TYPE ipfs_fsrepo_datastore_has_errors_total counter
ipfs_fsrepo_datastore_has_errors_total 0
# HELP ipfs_fsrepo_datastore_has_latency_seconds Latency distribution of Datastore.Has calls
# TYPE ipfs_fsrepo_datastore_has_latency_seconds histogram
ipfs_fsrepo_datastore_has_latency_seconds_bucket{le="0.0001"} 9
ipfs_fsrepo_datastore_has_latency_seconds_bucket{le="0.001"} 10
ipfs_fsrepo_datastore_has_latency_seconds_bucket{le="0.01"} 13
ipfs_fsrepo_datastore_has_latency_seconds_bucket{le="0.1"} 14
ipfs_fsrepo_datastore_has_latency_seconds_bucket{le="+Inf"} 14
ipfs_fsrepo_datastore_has_latency_seconds_sum 0.047101455
ipfs_fsrepo_datastore_has_latency_seconds_count 14
# HELP ipfs_fsrepo_datastore_has_total Total number of Datastore.Has calls
# TYPE ipfs_fsrepo_datastore_has_total counter
ipfs_fsrepo_datastore_has_total 14
# HELP ipfs_fsrepo_datastore_leveldb_delete_errors_total Number of errored Datastore.Delete calls
# TYPE ipfs_fsrepo_datastore_leveldb_delete_errors_total counter
ipfs_fsrepo_datastore_leveldb_delete_errors_total 0
# HELP ipfs_fsrepo_datastore_leveldb_delete_latency_seconds Latency distribution of Datastore.Delete calls
# TYPE ipfs_fsrepo_datastore_leveldb_delete_latency_seconds histogram
ipfs_fsrepo_datastore_leveldb_delete_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_leveldb_delete_latency_seconds_bucket{le="0.001"} 0
ipfs_fsrepo_datastore_leveldb_delete_latency_seconds_bucket{le="0.01"} 0
ipfs_fsrepo_datastore_leveldb_delete_latency_seconds_bucket{le="0.1"} 0
ipfs_fsrepo_datastore_leveldb_delete_latency_seconds_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_leveldb_delete_latency_seconds_sum 0
ipfs_fsrepo_datastore_leveldb_delete_latency_seconds_count 0
# HELP ipfs_fsrepo_datastore_leveldb_delete_total Total number of Datastore.Delete calls
# TYPE ipfs_fsrepo_datastore_leveldb_delete_total counter
ipfs_fsrepo_datastore_leveldb_delete_total 0
# HELP ipfs_fsrepo_datastore_leveldb_get_errors_total Number of errored Datastore.Get calls
# TYPE ipfs_fsrepo_datastore_leveldb_get_errors_total counter
ipfs_fsrepo_datastore_leveldb_get_errors_total 0
# HELP ipfs_fsrepo_datastore_leveldb_get_latency_seconds Latency distribution of Datastore.Get calls
# TYPE ipfs_fsrepo_datastore_leveldb_get_latency_seconds histogram
ipfs_fsrepo_datastore_leveldb_get_latency_seconds_bucket{le="0.0001"} 2
ipfs_fsrepo_datastore_leveldb_get_latency_seconds_bucket{le="0.001"} 2
ipfs_fsrepo_datastore_leveldb_get_latency_seconds_bucket{le="0.01"} 2
ipfs_fsrepo_datastore_leveldb_get_latency_seconds_bucket{le="0.1"} 2
ipfs_fsrepo_datastore_leveldb_get_latency_seconds_bucket{le="+Inf"} 2
ipfs_fsrepo_datastore_leveldb_get_latency_seconds_sum 7.4399e-05
ipfs_fsrepo_datastore_leveldb_get_latency_seconds_count 2
# HELP ipfs_fsrepo_datastore_leveldb_get_size_bytes Size distribution of retrieved byte slices
# TYPE ipfs_fsrepo_datastore_leveldb_get_size_bytes histogram
ipfs_fsrepo_datastore_leveldb_get_size_bytes_bucket{le="64"} 2
ipfs_fsrepo_datastore_leveldb_get_size_bytes_bucket{le="4096"} 2
ipfs_fsrepo_datastore_leveldb_get_size_bytes_bucket{le="262144"} 2
ipfs_fsrepo_datastore_leveldb_get_size_bytes_bucket{le="1.6777216e+07"} 2
ipfs_fsrepo_datastore_leveldb_get_size_bytes_bucket{le="+Inf"} 2
ipfs_fsrepo_datastore_leveldb_get_size_bytes_sum 68
ipfs_fsrepo_datastore_leveldb_get_size_bytes_count 2
# HELP ipfs_fsrepo_datastore_leveldb_get_total Total number of Datastore.Get calls
# TYPE ipfs_fsrepo_datastore_leveldb_get_total counter
ipfs_fsrepo_datastore_leveldb_get_total 2
# HELP ipfs_fsrepo_datastore_leveldb_has_errors_total Number of errored Datastore.Has calls
# TYPE ipfs_fsrepo_datastore_leveldb_has_errors_total counter
ipfs_fsrepo_datastore_leveldb_has_errors_total 0
# HELP ipfs_fsrepo_datastore_leveldb_has_latency_seconds Latency distribution of Datastore.Has calls
# TYPE ipfs_fsrepo_datastore_leveldb_has_latency_seconds histogram
ipfs_fsrepo_datastore_leveldb_has_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_leveldb_has_latency_seconds_bucket{le="0.001"} 0
ipfs_fsrepo_datastore_leveldb_has_latency_seconds_bucket{le="0.01"} 0
ipfs_fsrepo_datastore_leveldb_has_latency_seconds_bucket{le="0.1"} 0
ipfs_fsrepo_datastore_leveldb_has_latency_seconds_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_leveldb_has_latency_seconds_sum 0
ipfs_fsrepo_datastore_leveldb_has_latency_seconds_count 0
# HELP ipfs_fsrepo_datastore_leveldb_has_total Total number of Datastore.Has calls
# TYPE ipfs_fsrepo_datastore_leveldb_has_total counter
ipfs_fsrepo_datastore_leveldb_has_total 0
# HELP ipfs_fsrepo_datastore_leveldb_put_errors_total Number of errored Datastore.Put calls
# TYPE ipfs_fsrepo_datastore_leveldb_put_errors_total counter
ipfs_fsrepo_datastore_leveldb_put_errors_total 0
# HELP ipfs_fsrepo_datastore_leveldb_put_latency_seconds Latency distribution of Datastore.Put calls
# TYPE ipfs_fsrepo_datastore_leveldb_put_latency_seconds histogram
ipfs_fsrepo_datastore_leveldb_put_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_leveldb_put_latency_seconds_bucket{le="0.001"} 0
ipfs_fsrepo_datastore_leveldb_put_latency_seconds_bucket{le="0.01"} 0
ipfs_fsrepo_datastore_leveldb_put_latency_seconds_bucket{le="0.1"} 0
ipfs_fsrepo_datastore_leveldb_put_latency_seconds_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_leveldb_put_latency_seconds_sum 0
ipfs_fsrepo_datastore_leveldb_put_latency_seconds_count 0
# HELP ipfs_fsrepo_datastore_leveldb_put_size_bytes Size distribution of stored byte slices
# TYPE ipfs_fsrepo_datastore_leveldb_put_size_bytes histogram
ipfs_fsrepo_datastore_leveldb_put_size_bytes_bucket{le="64"} 0
ipfs_fsrepo_datastore_leveldb_put_size_bytes_bucket{le="4096"} 0
ipfs_fsrepo_datastore_leveldb_put_size_bytes_bucket{le="262144"} 0
ipfs_fsrepo_datastore_leveldb_put_size_bytes_bucket{le="1.6777216e+07"} 0
ipfs_fsrepo_datastore_leveldb_put_size_bytes_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_leveldb_put_size_bytes_sum 0
ipfs_fsrepo_datastore_leveldb_put_size_bytes_count 0
# HELP ipfs_fsrepo_datastore_leveldb_put_total Total number of Datastore.Put calls
# TYPE ipfs_fsrepo_datastore_leveldb_put_total counter
ipfs_fsrepo_datastore_leveldb_put_total 0
# HELP ipfs_fsrepo_datastore_leveldb_query_errors_total Number of errored Datastore.Query calls
# TYPE ipfs_fsrepo_datastore_leveldb_query_errors_total counter
ipfs_fsrepo_datastore_leveldb_query_errors_total 0
# HELP ipfs_fsrepo_datastore_leveldb_query_latency_seconds Latency distribution of Datastore.Query calls
# TYPE ipfs_fsrepo_datastore_leveldb_query_latency_seconds histogram
ipfs_fsrepo_datastore_leveldb_query_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_leveldb_query_latency_seconds_bucket{le="0.001"} 0
ipfs_fsrepo_datastore_leveldb_query_latency_seconds_bucket{le="0.01"} 0
ipfs_fsrepo_datastore_leveldb_query_latency_seconds_bucket{le="0.1"} 0
ipfs_fsrepo_datastore_leveldb_query_latency_seconds_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_leveldb_query_latency_seconds_sum 0
ipfs_fsrepo_datastore_leveldb_query_latency_seconds_count 0
# HELP ipfs_fsrepo_datastore_leveldb_query_total Total number of Datastore.Query calls
# TYPE ipfs_fsrepo_datastore_leveldb_query_total counter
ipfs_fsrepo_datastore_leveldb_query_total 0
# HELP ipfs_fsrepo_datastore_put_errors_total Number of errored Datastore.Put calls
# TYPE ipfs_fsrepo_datastore_put_errors_total counter
ipfs_fsrepo_datastore_put_errors_total 0
# HELP ipfs_fsrepo_datastore_put_latency_seconds Latency distribution of Datastore.Put calls
# TYPE ipfs_fsrepo_datastore_put_latency_seconds histogram
ipfs_fsrepo_datastore_put_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_put_latency_seconds_bucket{le="0.001"} 0
ipfs_fsrepo_datastore_put_latency_seconds_bucket{le="0.01"} 0
ipfs_fsrepo_datastore_put_latency_seconds_bucket{le="0.1"} 0
ipfs_fsrepo_datastore_put_latency_seconds_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_put_latency_seconds_sum 0
ipfs_fsrepo_datastore_put_latency_seconds_count 0
# HELP ipfs_fsrepo_datastore_put_size_bytes Size distribution of stored byte slices
# TYPE ipfs_fsrepo_datastore_put_size_bytes histogram
ipfs_fsrepo_datastore_put_size_bytes_bucket{le="64"} 0
ipfs_fsrepo_datastore_put_size_bytes_bucket{le="4096"} 0
ipfs_fsrepo_datastore_put_size_bytes_bucket{le="262144"} 0
ipfs_fsrepo_datastore_put_size_bytes_bucket{le="1.6777216e+07"} 0
ipfs_fsrepo_datastore_put_size_bytes_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_put_size_bytes_sum 0
ipfs_fsrepo_datastore_put_size_bytes_count 0
# HELP ipfs_fsrepo_datastore_put_total Total number of Datastore.Put calls
# TYPE ipfs_fsrepo_datastore_put_total counter
ipfs_fsrepo_datastore_put_total 0
# HELP ipfs_fsrepo_datastore_query_errors_total Number of errored Datastore.Query calls
# TYPE ipfs_fsrepo_datastore_query_errors_total counter
ipfs_fsrepo_datastore_query_errors_total 0
# HELP ipfs_fsrepo_datastore_query_latency_seconds Latency distribution of Datastore.Query calls
# TYPE ipfs_fsrepo_datastore_query_latency_seconds histogram
ipfs_fsrepo_datastore_query_latency_seconds_bucket{le="0.0001"} 0
ipfs_fsrepo_datastore_query_latency_seconds_bucket{le="0.001"} 0
ipfs_fsrepo_datastore_query_latency_seconds_bucket{le="0.01"} 0
ipfs_fsrepo_datastore_query_latency_seconds_bucket{le="0.1"} 0
ipfs_fsrepo_datastore_query_latency_seconds_bucket{le="+Inf"} 0
ipfs_fsrepo_datastore_query_latency_seconds_sum 0
ipfs_fsrepo_datastore_query_latency_seconds_count 0
# HELP ipfs_fsrepo_datastore_query_total Total number of Datastore.Query calls
# TYPE ipfs_fsrepo_datastore_query_total counter
ipfs_fsrepo_datastore_query_total 0
# HELP ipfs_p2p_peers_total Number of connected peers
# TYPE ipfs_p2p_peers_total gauge
ipfs_p2p_peers_total{transport="/ip4/tcp"} 4
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 0.18
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 2048
# HELP process_open_fds Number of open file descriptors.
# TYPE process_open_fds gauge
process_open_fds 24
# HELP process_resident_memory_bytes Resident memory size in bytes.
# TYPE process_resident_memory_bytes gauge
process_resident_memory_bytes 2.7758592e+07
# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1.49935471991e+09
# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 3.89312512e+08

```

</details>

**Comment by diasdavid** [comment]
@VictorBjelkholm check out http://npmjs.org/prom-client, most of that templating comes from there, it is a good head start :)

**Comment by VictorBjelkholm** [comment]
Now (https://github.com/ipfs/js-ipfs/pull/913) we have the possibility of getting metrics from a running js-ipfs node! Docs here:  https://github.com/ipfs/js-ipfs#monitoring

Created another issue in the dweblink-infra repo for tracking work on getting some js-ipfs gateways up and running: https://github.com/ipfs/dweblink-infra/issues/1

Then we'll be full-speed ahead serving content to the galaxy from js-ipfs 🎉

**Comment by diasdavid** [comment]
Really close to https://github.com/ipfs/js-ipfs/pull/968/files

Can we have `js.ipfs.io` as a bunch of js-ipfs gateway nodes, @lgierth @VictorBjelkholm ??

**Comment by VictorBjelkholm** [comment]
@diasdavid 
> Can we have js.ipfs.io as a bunch of js-ipfs gateway nodes

Yeah, sounds good to me. Perhaps could be more specific, like `js-gateway.ipfs.io`, as we probably want to have bootstrap peers under similar dns as well.

**Comment by diasdavid** [comment]
I like `js.ipfs.io` more.

**Comment by diasdavid** [comment]
Ok, Gateway has been merged into master. @VictorBjelkholm can we have just one js-ipfs node as a gateway under js.ipfs.io? @lgierth ? 🙏🏽 please!

**Comment by diasdavid** [comment]
Hi @lgierth you asked me two weeks ago to ping you in two weeks if this hasn't moved. Any update here?

**Comment by lgierth** [comment]
Not getting to this over the rest of this year -- this should be done on the new infra and that's not ready yet.

**Comment by diasdavid** [comment]
😢

What's the new infra?

**Comment by diasdavid** [comment]
@VictorBjelkholm @lgierth any love on this one?

**Comment by VictorBjelkholm** [comment]
@diasdavid I tried picking this up before but after hitting https://github.com/ipfs/js-ipfs/issues/1221, I got stuck.

I feel like we have to go through the daemon and make sure there is no errors that will crash the entire process, should be able to handle some errors without exit before we can ship this.

**Comment by VictorBjelkholm** [comment]
Few things missing for stable deploys (I'll update this with more things once I come across them)

- [ ] Ability to set Announce/NoAnnounce for swarm addresses
- [ ] Less crashy daemon (fetching something then cancelling crashes the daemon)
- [ ] More metrics exposed on the metrics endpoint

**Comment by diasdavid** [comment]
> Less crashy daemon (fetching something then cancelling crashes the daemon)

Can you add a test for this? It shouldn't be the case at all.

**Comment by VictorBjelkholm** [comment]
> Can you add a test for this? It shouldn't be the case at all.

Will do, but that was just a example. Main issue is that the daemon even can crash when something lower-level is throwing. We don't seem to have the error boundary correct in the daemon. No error (unless daemon issues) should crash it.

**Comment by diasdavid** [comment]
Agreed. The whole point of this exercise is to see how the js-ipfs daemon behaves as a gateway and over a long period of time to capture issues that aren't capture in tests. I'm glad that we are already finding them :)

**Comment by VictorBjelkholm** [comment]
- [ ] Daemon crashes if webrtc/ws star-signal server is down

**Comment by diasdavid** [comment]
@VictorBjelkholm that's expected, I believe (depends on the log message). If the daemon can't start a listener, it won't start (and therefore crash and stop).

**Comment by VictorBjelkholm** [comment]
Hm, I understand what you're saying. But if my local state is exactly the same, I don't expect the daemon to start crashing because of some external application. I think it would be more reasonable to output a warning/error but keep the daemon running.

**Comment by diasdavid** [comment]
@VictorBjelkholm just saw the IPFS All Hands notes

<img width="624" alt="image" src="https://user-images.githubusercontent.com/1211152/37031941-17fce32a-2141-11e8-96bb-f9ba9a43cbc0.png">

Woowow!! 👏🏽

Can you put it accessible behind `js.ipfs.io`, pretty please?

**Comment by VictorBjelkholm** [comment]
@diasdavid indeed :) It's coming, waiting for Lars to verify the DNS change and to review the infra change.

**Comment by diasdavid** [comment]
@VictorBjelkholm js.ipfs.io is not letting me using the js-ipfs daemon as a gateway -- https://js.ipfs.io/ipfs/QmSXArjcBFsHY7WznMyuAqZKVh5cX95DkeNyR5wqcNmprQ

**Comment by VictorBjelkholm** [comment]
@diasdavid missed this. Two points: daemon is too unstable to rely on currently, so I've stopped working on it for now. Second is that that I haven't yet setup the DNS records for js.ipfs.io, that's why it's not working for you :)

**Comment by fsdiogo** [comment]
@VictorBjelkholm I'm going to work on having the daemon more stable, as in https://github.com/ipfs/js-ipfs/issues/1325.

The thing is: I'm not sure how to tackle this, as you're working on it too can you point me in the right direction?

**Comment by SgtPooki** [comment]
js-ipfs is being deprecated in favor of [Helia](https://github.com/ipfs/helia). You can https://github.com/ipfs/js-ipfs/issues/4336 and [read the migration guide](https://github.com/ipfs/helia/wiki/%F0%9F%9A%9B-Migrating-from-js-IPFS).

Please feel to reopen with any comments by 2023-06-02. We will do a final pass on reopened issues afterwards (see https://github.com/ipfs/js-ipfs/issues/4336).

@achingbrain @BigLep do we have js-ipfs running on infra? Do we have a tracking item for replacing them with Helia?

**Comment by BigLep** [comment]
js-ipfs is not running in NodeJS context on any PL-owned infra that I'm aware of.

**Comment by achingbrain** [comment]
No, there were big plans that never really got executed.

Closing as this was a piece of work that never got finished.
